master/internal/agent/agent.go

```diff
@@ -68,7 +96,31 @@ func (a *agent) Receive(ctx *actor.Context) error {
 		if err := ctx.Ask(a.socket, wsm).Error(); err != nil {
 			ctx.Log().WithError(err).Error("failed to write master set agent options")
 		}
+		if a.awaitingReconnect {
+			ctx.Log().Info("agent reconnected")
+			a.awaitingReconnect = false
+
+			// Re-propagate our old state back on successful recovery.
+			if a.enabled {
+				ctx.Tell(a.resourcePool, sproto.EnableAgent{Agent: ctx.Self()})
+			} else {
+				ctx.Tell(a.resourcePool, sproto.DisableAgent{Agent: ctx.Self(), Drain: a.draining})
```

Comment:
So the draining and enabled states here start to differ from the same state in resource pool's structures. We used to have a duplicated state (not accounting for latency between actors passing messages), and now it'll actually be different. I think this adds unnecessary complexity, and I don't like that, I'd store the "previous" state in a separate fields. But I guess we can live with that for now.

Reply:
yeah, earlier in this PR i had an agent state stack, where i pushed the this temp drain on top then pop it off on reconnect (so the state vars were always accurate). i will add that back, i agree this is 'tricky'.